src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -88,6 +79,16 @@ def _post(self, request: Request):
         return Response(response, status=200)
 
     def _should_use_v3(self, version, request):
+        """Determine whether the v3 computation should be used for project configs.
+
+        - When v4 is requested, the project config should have the same behavior
```

Comment:
Should the tags in this function be update to `relay_use_v4` when the version 4 is actually requested? Seems like the right thing to do.

Reply:
Not sure what you mean. The tag `relay_endpoint_version` should have the version automatically set for v4, and I don't think we should rename `relay_use_v3` because the computation is the same for both versions (I don't see any benefit). Is there something else I'm missing?

Reply:
We should rename this function. It's not about using v3, it's more like a "should use anything newer than v2 for project configs". v3 and v4 are identical when it comes to project configs, their only difference is in supporting global configs. So, really this function checks whether v3+ can be used.

If you like, you can think about this in a simplified way, and also adjust the documentation accordingly: When partial configs are requested, project config computation must fall back to v2 implementation.

Reply:
I renamed the function to `_should_full_build_projconfig` but tbqh I don't really like the name, open for suggestions.

The only v3 reference left is in the metrics, and I'm hesitant to change it to not break any active monitoring; I added a comment to reflect it. Also open to update it if you guys think it's a good call!